src/ring/middleware/session_timeout.clj

```diff
@@ -4,6 +4,22 @@
 (defn- current-time []
   (quot (System/currentTimeMillis) 1000))
 
+(defn- session-idle-expired? [session]
+  (let [end-time (::idle-timeout session)]
+    (and end-time (< end-time (current-time)))))
+
+(defn- session-absolute-expired? [session]
+  (let [end-time (::absolute-timeout session)]
+    (and end-time (< end-time (current-time)))))
+
+(defn- idle-session-timeout-response [req-session response timeout]
```

Comment:
Can you make the `response` argument come first? Same with the other `-response` function. This is more consistent with Ring conventions.